packages/editor/patches/@tiptap+extension-list-keymap+2.6.6.patch

```diff
@@ -13,8 +13,20 @@ index 0f8ae36..2f715b5 100644
          const $listPos = editor.state.doc.resolve($anchor.before() - 1);
          const listDescendants = [];
          $listPos.node().descendants((node, pos) => {
+@@ -107,6 +109,11 @@ const handleBackspace = (editor, name, parentListTypes) => {
+     if (!core.isAtStartOfNode(editor.state)) {
+         return false;
+     }
++    // if previous item is not a list (could be a paragraph in the current list item)
++    // join to the same list item
++    if (!hasListItemBefore(name, editor.state)) {
```

Comment:
After some more testing, this is not going to work. It breaks in cases where you are backspacing in the first item of the list. This logic should be more robust to handle only the cases where we are backspacing in the second or later child inside a list item.

I also recommend adding tests for other cases:

1. Backspacing at the start of the first list item
2. Backspacing when there is a list right before the list item

It should handle all these cases correctly.